test/WebJobs.Script.Tests.Shared/TempDirectory.cs

```diff
@@ -0,0 +1,52 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+using System.IO;
+
+namespace WebJobs.Script.Tests
+{
+    public sealed class TempDirectory : IDisposable
+    {
+        public TempDirectory()
```

Comment:
This is a new test helper that makes it simpler to deal with short lived temp folders in tests, ensuring we don't leave stuff behind.